internal/access/credential.go

```diff
@@ -72,9 +72,24 @@ func UpdateCredential(c *gin.Context, user *models.Identity, newPassword string)
 	return nil
 }
 
-func updateCredential(c *gin.Context, user *models.Identity, newPassword string, isSelf bool) error {
+func updateCredential(c *gin.Context, user *models.Identity, oldPassword, newPassword string, isSelf, requireOldPassword bool) error {
 	db := getDB(c)
 
+	// Users have to supply their old password to change their existing password
+	if isSelf && requireOldPassword {
```

Comment:
Should we move this check to a wrapper function rather than adding another boolean flag to the function parameters? 

Ex:
```
func changePassword(c *gin.Context, user *models.Identity, oldPassword, newPassword string, isSelf) {
  if isSelf {
    // require old password
  }
  updateCredential(c, user, newPassword, isSelf)
}
func updateCredential(c *gin.Context, user *models.Identity, oldPassword, newPassword string, isSelf) {
  // do the actual updating directly here
}
```

Reply:
Ya, seems like we can remove both the `oldPassword` and `requireOldPassword` arguments by doing this check in `access.UpdatePassword`.

Reply:
Fixed.